server/graphql/v2/mutation/MemberMutations.ts

```diff
@@ -31,7 +32,88 @@ const isLastAdmin = async (account, memberAccount) => {
   return admins.length === 1 && admins[0].MemberCollectiveId === memberAccount.id;
 };
 
+const GraphQLFollowResult = new GraphQLObjectType({
+  name: 'FollowResult',
+  fields: {
+    individual: {
+      type: new GraphQLNonNull(GraphQLIndividual),
+    },
+    member: {
+      type: new GraphQLNonNull(GraphQLMember),
+    },
+  },
+});
+
+const GraphQLUnfollowResult = new GraphQLObjectType({
+  name: 'UnfollowResult',
+  fields: {
+    individual: {
+      type: new GraphQLNonNull(GraphQLIndividual),
+    },
+    member: {
+      type: GraphQLMember,
+    },
+  },
+});
+
 const memberMutations = {
+  follow: {
+    type: new GraphQLNonNull(GraphQLFollowResult),
+    description: 'Follows a given Collective. Scope: "account"',
+    args: {
+      account: {
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+        description: 'user will follow this account',
+      },
+    },
+    async resolve(_, args, req: Express.Request) {
+      checkRemoteUserCanUseAccount(req);
+
+      const account = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+
+      const [member] = await Member.findOrCreate({
+        where: {
+          MemberCollectiveId: req.remoteUser.CollectiveId,
+          CollectiveId: account.id,
+          role: MemberRoles.FOLLOWER,
+        },
+        defaults: {
+          CreatedByUserId: req.remoteUser.id,
+        },
+      });
```

Comment:
We should either:
- Rate limit the endpoint
- Or make sure that we're re-using any previously deleted membership, settings its `deletedAt` to null

In the current state, someone spamming the _Follow/Unfollow_ button would create many entries in the DB.